### PR TITLE
Update smbj and bouncy castle to support SMBv3

### PR DESCRIPTION
**Pull request by ekelly-1898**
## Description
Upgrade SMBJ to support SMBv3.

## Motivation and Context
Explanation of why these changes are being proposed, including any links to other relevant issues or pull requests.

## How Has This Been Tested?
Integration tests succeed.

### CODE REVIEW

**Review by compliance-r7 — DISMISSED** [review]
Only pom.xml has changed, approved!

**Review by mgrundie-r7 — DISMISSED** [review]

**Review by rkirk-r7 — DISMISSED** [review]

File: src/main/java/com/rapid7/client/dcerpc/dto/ContextHandle.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ public static ContextHandle fromHex(final String hString) {
         if (hString == null || hString.length() > (bytes.length * 2)) {
             throw new IllegalArgumentException("hString is invalid: " + hString);
         }
-        final byte[] handle = Hex.decode(hString);
+        final byte[] handle = BaseEncoding.base16().decode(hString.toUpperCase());
```
Comment (rkirk-r7) [inline src/main/java/com/rapid7/client/dcerpc/dto/ContextHandle.java:79]: Does it require being upper case to decode then? As we are not obviously uppercasing in the other usages

File: src/main/java/com/rapid7/client/dcerpc/msrrp/dto/RegistryValue.java [hunk 2]
```diff
@@ -132,7 +135,7 @@ public String getDataAsBinaryStr() {
     }
 
     public String getDataAsHexStr() {
-        return Hex.toHexString(data).toUpperCase();
+        return BaseEncoding.base16().encode(data); //Hex.toHexString(data).toUpperCase();
```
Comment (rkirk-r7) [inline src/main/java/com/rapid7/client/dcerpc/msrrp/dto/RegistryValue.java:138]: remove comment

Should this still uppercase the encoded data or is that defaulted to?

**Review by rkirk-r7 — COMMENTED** [review]
looks good to me although struggling to follow the uppercasing requirements

File: src/main/java/com/rapid7/client/dcerpc/msrrp/dto/RegistryValue.java (hunk 2 above)
Comment (pwatson-r7) [reply src/main/java/com/rapid7/client/dcerpc/msrrp/dto/RegistryValue.java:138]: it defaults to uppercase
[Base Encoding](https://guava.dev/releases/16.0/api/docs/com/google/common/io/BaseEncoding.html)

File: src/main/java/com/rapid7/client/dcerpc/dto/ContextHandle.java (hunk 1 above)
Comment (pwatson-r7) [reply src/main/java/com/rapid7/client/dcerpc/dto/ContextHandle.java:79]: The BaseEncoding encodes and decodes using uppercase characters by default.

**Review by dbeliakov-r7 — APPROVED** [review]

**Pull request merged**